src/metaxy/metadata_store/duckdb.py

```diff
@@ -165,3 +226,49 @@ def _create_native_components(self):
         diff_resolver = NarwhalsDiffResolver()
 
         return joiner, calculator, diff_resolver
+
+    # ------------------------------------------------------------------ DuckLake
+    def open(self) -> None:
+        """Open DuckDB connection and configure optional DuckLake attachment."""
+        super().open()
+        if self._ducklake_attachment is not None:
+            duckdb_conn = self._duckdb_raw_connection()
+            self._ducklake_attachment.configure(duckdb_conn)
+
+    def preview_ducklake_sql(self) -> list[str]:
+        """Return DuckLake attachment SQL if configured."""
+        return self.ducklake_attachment.preview_sql()
+
+    @property
+    def ducklake_attachment(self) -> DuckLakeAttachmentManager:
+        """DuckLake attachment manager (raises if not configured)."""
+        if self._ducklake_attachment is None:
+            raise RuntimeError("DuckLake attachment is not configured.")
+        return self._ducklake_attachment
+
+    @property
+    def ducklake_attachment_config(self) -> DuckLakeAttachmentConfig:
+        """DuckLake attachment configuration (raises if not configured)."""
+        if self._ducklake_config is None:
+            raise RuntimeError("DuckLake attachment is not configured.")
+        return self._ducklake_config
+
+    def _duckdb_raw_connection(self) -> DuckDBConnection:
+        """Return the underlying DuckDBPyConnection from the Ibis backend."""
+        if self._conn is None:
+            raise RuntimeError("DuckDB connection is not open.")
+
+        backend = self._conn
+        candidate = getattr(backend, "con", None)
```

Comment:
Should use type narrowing with `isinstance` checks and raise a clear error for unexpected types. 
Right now it's impossible what types except `DuckDBConnection` is this code trying to handle